src/tracing.rs

```diff
@@ -75,6 +98,27 @@ pub fn trace_set(level: TraceLevel, cb: TracingCb) -> bool {
     return true;
 }
 
+/// Passes [trace_set] a shim function to pass tracing info to the [log] crate.
+pub fn trace_shim_log_crate() {
```

Comment:
While this shim is nice, I don't think every git2 user needs this. git2, from my observation [^1], has tried to be a thin layer upon libgit2. This kind of glue code is pretty application facing, and IMO should leave for users to decide. 

[^1]: I am not really a so-called long-term maintainer